src/Chapter3.hs

```diff
@@ -907,9 +1108,27 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+data Gold = Gold
+  { gold  :: Int
+  } deriving (Show)
+
 class Append a where
     append :: a -> a -> a
 
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append x y = Gold((gold x) + (gold y))
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append x y = concat [x, y]
+
+instance (Append a) => Append (Maybe a) where
+    append :: Maybe a -> Maybe a -> Maybe a
+    append (Just x) (Just y) = Just (append x y)
+    append Nothing (Just y) = Just y
+    append (Just x) Nothing = Just x
+    append Nothing Nothing = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂